src/components/Billing/Invoices.tsx

```diff
@@ -2,18 +2,64 @@ import { Title } from '@mantine/core'
 import CustomTable from 'components/common/CustomTable'
 import { PrintModal } from 'components/common/Modals'
 import { useDisclosure } from '@mantine/hooks'
-import { useCallback } from 'react'
+import { useCallback, useMemo, useState } from 'react'
 
-import { invoiceElements } from './mockedData'
+import useCampaignsData from 'hooks/useCampaignsData'
+// TODO: Delete mock data
+// import { invoiceElements } from './mockedData'
+import { CampaignFundsActive, IInvoices } from 'types'
+import dayjs from 'dayjs'
+import { CampaignStatus } from 'adex-common'
+import useAccount from 'hooks/useAccount'
 
 const columnTitles = ['Company Name', 'Campaign Period', 'Amount Spent']
 
 const Invoices = () => {
   const [opened, { open, close }] = useDisclosure(false)
+  const { campaignsData } = useCampaignsData()
+  const campaigns = useMemo(() => Array.from(campaignsData.values()), [campaignsData])
+  const {
+    adexAccount: {
+      billingDetails: { companyName },
+      fundsOnCampaigns: { perCampaign }
+    }
+  } = useAccount()
+
+  const [selectedCampaignId, setSelectedCampaignId] = useState('')
+
+  const getCurrencyName = useCallback(
+    (campaignId: string, campaings: CampaignFundsActive[]) =>
+      campaignId && !!campaings.length
+        ? campaings.find((item) => item.id === campaignId)?.token.name || ''
+        : '',
+
+    []
+  )
+
+  const invoiceElements: IInvoices[] = useMemo(
+    () =>
+      campaigns
+        .filter(
+          (c) =>
+            c.campaign.status === CampaignStatus.expired ||
```

Comment:
`[ CampaignStatus.expired, CampaignStatus.closedByUser, c.campaign.status === CampaignStatus.exhausted].includes( c.campaign.status)`

it can be const